### ISSUE TITLE
[Programming question]: Sandboxed code execution

### ISSUE THREAD

**Issue by haesleinhuepf** [haesleinhuepf/git-bob-playground#177]
Let's take the code from this issue: https://github.com/haesleinhuepf/git-bob/issues/443 and put it in a new notebook. Before running this, store 5 in the environment variable "HELLO". Within the remotely executed code, read that environment variable and print it out. Also print out the environment variable in the main code, after the execution of the sandboxed code.

git-bob try to do this

**Comment by haesleinhuepf** [comment]
Obviously, the sandbox can read from my environment variables. How can I configure the sandbox in a way that it cannot do this?

git-bob answer

**Comment by haesleinhuepf** [comment]
git-bob try this out!

**Comment by shleder** [comment]
For the HELLO test: pass the variable explicitly across the sandbox boundary rather than inheriting the environment. Inheriting environ leaks whatever happens to be set (keys, tokens); an explicit allowlist (`HELLO` in, everything secret-shaped out) makes the test deterministic and the boundary auditable. Disclosure: I build [vetto](https://github.com/shleder/vetto), a daemon-less kernel sandbox for AI coding agents (Landlock/seccomp on Linux, Seatbelt on macOS) — mentioning it only because it's directly relevant here. — we deny secret-shaped env names by default for this reason.
